pingcap/tidb-operator
# improve performance of cleanup BR

### What problem does this PR solve?

Close #4073
* improve performance of cleanup br
* add e2e test for cleanup br

### What is changed and how does it work?


### Code changes


### Tests


1. Deploy `Backup` with clean policy 'Delete'.
2. Add files to bucket as backup files.
2. Delete `Backup`.
3. Watch log, and check if all files in storage are deleted.
* AWS S3: 10000 files takes 24 seconds
* NFS: 10000 files takes 17 seconds
* GCS: 10000 files takes 43 seconds

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Improve performance of cleanup `BR`
```
